deploy.rb

```diff
@@ -377,6 +383,17 @@
   end
 end
 
+if CREATE_AND_PUSH_BRANCH
+  in_step Step::CREATE_AND_PUSH_BRANCH do
```

Comment:
I think we should do that after a successful deploy. Else, it probably means that the generated files are wrong.

Reply:
Sounds good to me

Reply:
Oh, you also only want to do that if `GIT_REPO` has been specified. This might sound annoying, but our tests won't always rely on the git repo being present.